dissect/formats/mbr.py

```diff
@@ -0,0 +1,119 @@
+'''
+Structures useful for parsing an MS-DOS style MBR.
+'''
+import os
+import math
+import logging
+import contextlib
+import collections
+
+import vstruct.types as v_types
+
+
+# assumption: sectors are 512-bytes in lenght
+# anything other than this is... non-standard, and weird.
+SECTOR_SIZE = 512
+
+
+# partition types
+SYSTEMID = v_types.venum()
+SYSTEMID.EMPTY           = 0
+SYSTEMID.FAT_12       = 1
+SYSTEMID.XENIX_ROOT      = 2
+SYSTEMID.XENIX_USR       = 3
+SYSTEMID.FAT_16_INF32MB  = 4
+SYSTEMID.EXTENDED        = 5
+SYSTEMID.FAT_16          = 6
+SYSTEMID.NTFS_HPFS       = 7
+SYSTEMID.AIX             = 8
+SYSTEMID.AIX_BOOT        = 9
+SYSTEMID.OS2_BOOT_MGR    = 10
+SYSTEMID.PRI_FAT32_INT13 = 11
+SYSTEMID.EXT_FAT32_INT13 = 12
+SYSTEMID.EXT_FAT16_INT13 = 14
+SYSTEMID.WIN95_EXT       = 15
+SYSTEMID.OPUS            = 16
+SYSTEMID.FAT_12_HIDDEN   = 17
+SYSTEMID.COMPAQ_DIAG     = 18
+SYSTEMID.FAT_16_HIDDEN_INF32MB = 20
+SYSTEMID.FAT_16_HIDDEN   = 22
+SYSTEMID.NTFS_HPFS_HIDDEN= 23
+SYSTEMID.VENIX           = 64
+SYSTEMID.NOVEL0          = 81
+SYSTEMID.MICROPORT       = 82
+SYSTEMID.GNU_HURD        = 99
+SYSTEMID.NOVEL1          = 100
+SYSTEMID.PC_IX           = 117
+SYSTEMID.MINUX_OLD       = 128
+SYSTEMID.MINUX_LINUX     = 129
+SYSTEMID.LINUX_SWAP      = 130
+SYSTEMID.LINUX_NATIVE    = 131
+SYSTEMID.AMOEBA          = 147
+SYSTEMID.AMOEBA_BBT      = 148
+SYSTEMID.BSD_386         = 165
+SYSTEMID.BSDI_FS         = 183
+SYSTEMID.BSDI_SWAP       = 184
+SYSTEMID.SYRINX          = 199
+SYSTEMID.CP_M            = 219
+SYSTEMID.ACCESS_DOS      = 225
+SYSTEMID.DOS_R_O         = 227
+SYSTEMID.DOS_SECONDARY   = 242
+SYSTEMID.BBT             = 255
+
+
+# partition boot flag
+BOOTINDICATOR = v_types.venum()
+BOOTINDICATOR.NOBOOT = 0
+BOOTINDICATOR.SYSTEM_PARTITION = 128
+
+
+class PART_ENTRY(v_types.VStruct):
+    '''
+    partition entry in the MBR.
+    '''
+    def __init__(self):
+        super(PART_ENTRY, self).__init__()
```

Comment:
So, the thing that complicates this is if *any* of the classes you multi-inherit from actually need arguments to their constructor ( common in most of my code, but not as much with vstructs :) ) you can't really blindly call the ctor chain with no args anyway.  Which then means in all those cases, it'll need to be broken out explicitly.  While i totally dig the DRY concepts (how could one not) i'm also not a huge fan of having 2/3 different context-decided conventions for how things "should be" in the codebase ( trying hard to make sure all the code in the various projects uses the same conventions for readability ).  I guess if you know that you need not pass any args to your parent constructor, using super is fine.  It's really only 2 conventions.

1. if you pass args to your parent ctors, do not use super
2. if not, use <pre>super(ClassFoo,self).__init__()</pre> since it must be py27 / py34+ compat.

Mostly just wanted to make sure that if we're going to use it, we discuss it :)